### ISSUE TITLE
About loss R2

### ISSUE THREAD

**Issue by EricLingRui** [yueyu1030/COSINE#2]
Thank you for your contribution，when I read the code, I can't understand this line. 
https://github.com/yueyu1030/COSINE/blob/2b735e4dbbb1a7231e407ebb0566d921da4f04e1/trainer.py#L75
Is this R2 mentioned in the paper?  
why not
`loss = nn.KLDivLoss(reduction = 'none')`
`loss_r2 = loss(input, 1/n_classes_)`
`l -= confreg *( torch.sum(loss_r2 * w.unsqueeze(1))`

**Comment by Janeey99** [comment]
I have same problem with you!!! Did you understand???
